packages/website/blog/2022-12-23-consistent-type-exports-and-imports-why-and-how.md

```diff
@@ -0,0 +1,97 @@
+---
+authors:
+  - image_url: https://www.joshuakgoldberg.com/img/josh.jpg
+    name: Josh Goldberg
+    title: typescript-eslint Maintainer
+    url: https://github.com/JoshuaKGoldberg
+description: Why enforcing TypeScript imports use the `type` modifier when possible benefits some project setups.
+slug: consistent-type-exports-and-imports-why-and-how
+tags: [typescript, imports, exports, types, transpiling]
+title: 'Consistent Type Exports and Imports: Why and How'
+---
+
+TypeScript 3.8 [added type-only imports and exports](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-8.html):
+
+```ts
+import type { SomeThing } from './some-module.js';
+export type { SomeThing };
+```
+
+The key difference with `export type` and `import type` is that they are _only in the type system_.
+Attempting to use a _value_ imported as only a _type_ in runtime code will cause a TypeScript error:
+
+```ts twoslash
+import type { SomeThing } from './some-module.js';
+
+new SomeThing();
+//  ~~~~~~~~~
+// 'SomeThing' cannot be used as a value
+// because it was imported using 'import type'.
+```
+
+This brings up two questions:
+
+- Why would you want to use these type-only imports and exports?
+- How can you enforce a project use them whenever necessary?
+
+Let's Dig In!
+
+<!--truncate-->
+
+## Functional Benefits
+
+Some modules in code may cause _side effects_: code that is run when the module is imported and causes changes outside the module.
```

Comment:
```suggestion
### Module Side Effects

Some modules in code may cause _side effects_: code that is run when the module is imported and causes changes outside the module.
```

May be good to add sub-headings here to break this up and make it clear that these are two separate benefits?

Otherwise (to me at least) it sort of feels like they run together.